### DISCUSSION TITLE
Upcoming wallabag version 3.0.0 & 2.6.0

### DISCUSSION THREAD

**Discussion by j0k3r** [wallabag/wallabag#6177] [Announcements]
Hello everyone,

The latest version of wallabag is using [Symfony v3.4.49](https://symfony.com/releases/3.4) which hasn't been maintained since November 2020 and without security fixes since November 2021.

@yguedidi started to work on migrating/updating some parts of wallabag to remove deprecations, update services and many other changes in the code base in order to be able to upgrade Symfony.
Right now, the master of wallabag is running on Symfony v4.4.49 and we'll soon upgrade to Symfony 5.

While moving to Symfony 5 we also wanted to raise the minimum PHP version from 7.4 to 8.0.

We are now working on moving to [Symfony Flex](https://github.com/symfony/flex) which "_is so convenient and improves your productivity so much that it's strongly recommended_". The main change of Symfony Flex is the directory structure.
It changes completely from the previous directory structure we are currently using:
- assets moved in a different directory
- config files are exploded by packages
- parameters are now defined in an environment file
- there are no bundles anymore (we were using AnnotationBundle, ApiBundle, CoreBundle, ImportBundle & UserBundle)
- templates are now in a single place, like translations & migrations
- etc.

We were planning to push all these upgrades (Flex & Symfony 5 & PHP 8) to the version 2.6.0.
But the more we are working on that technical upgrade the more we think it will be a complex upgrade for people running a wallabag instance mostly because of the directory structure change.

### 3.0.0
That's why we decided to put all these technical upgrades into a new breaking change version: **3.0.0**.

- For people using wallabag as a service (like those who have a subscription plan on https://www.wallabag.it/en (thanks you 🫶)), they won't notice a single change. There won't be functional change on how they use wallabag.
- For people hosting wallabag, it'll be an upgrade more significant than just running `make update`.

We would like to put these technical changes into 3.0.0:
- Symfony 6
- PHP 8.1 or higher
- new directory structure

Of course, we will recommend to everyone to upgrade to 3.0.0 as soon as the version is released but we'll also have a 2.x branch where we might still provide some bug fix for people who don't want to upgrade right now. But keep in mind that as soon as the 3.0 will be released, we'll be more focused on that version than on the older one.

Some features planned for 3.0 will be postponed to 4.0 others will be for 3.1.0.

### 2.6.0
In the coming days (or maybe weeks) we'll release the 2.6.0 which will upgrade Symfony from 3.4 to 4.4.
It'll resolve a lot of bugs for people running wallabag on PHP 8 for example.
The required PHP version will stay the same as before: 7.4 or higher.

**Comment by Strubbl** [comment]
Debian stable still uses php 7.4, see https://packages.debian.org/bullseye/php.

**Reply by j0k3r** [reply]
I'm sorry for people using Debian stable and willing to use wallabag 3.0.0.
They will still be able to use wallabag 2.6.0.

We won't stop the upgrade to Symfony 6 & PHP 8.1 and have a more faster and up to date wallabag **because** of Debian 🤷🏼

**Reply by j0k3r** [reply]
Debian Bookworm (probably released next summer) will have PHP 8.1, see https://packages.debian.org/bookworm/php
So we are "just" 6 months ahead.

**Reply by Strubbl** [reply]
> We won't stop the upgrade to Symfony 6 & PHP 8.1 and have a more faster and up to date wallabag **because** of Debian 🤷🏼

Yes, of course not. Just wanted to mention this.

**Comment by KhaaL** [comment]
Thanks for the update, excited about the 3.0 update. Good thing I haven't taken the plunge to selfhost it yet 😄

Is there any plan for front-end features? Tag management is painful to use atm from the webbrowser.

**Reply by j0k3r** [reply]
Once the 3.0 will be released, I hope we'll find time to improve/add some features on the front-end side!

**Comment by formlessdao** [comment]
Can you write please non-developer friendly instructions so it can run locally on any Windows or Linux PC?
I just want to run it on my own PC not on some server.

**Comment by romu70** [comment]
Hi guys.
Is there plans to upgrade the old Ubuntu 16.04 the Docker image is still based on?

**Reply by j0k3r** [reply]
What Docker image are you talking about? Both the one from the wallabag/wallabag & wallabag/docker image are based on alpine not ubuntu.

**Reply by romu70** [reply]
I didn't notice this has changed. Initially, the image was based on Ubuntu 16.04, right? And as I still see this Ubuntu image on my server, I was asking. If the image has already been rebased on Alpine, that's perfect.

Thanks.

**Reply by j0k3r** [reply]
It's weird because the official image is based on alpine since the beginning.
The dev image (inside the main repo) is based on `php:8.1-fpm` which use Debian. Maybe you are talking about that one 🤷‍♂️

**Reply by romu70** [reply]
Ok, so the problem is on my side. Maybe it comes from when I created my own image before the official one was created. Dunno, it's ooolllldddddd. Anyway, thanks, I've a way to cleanup my nas.

**Comment by dmhowcroft** [comment]
I've done some poking around, but I can't find any information on progress for the 3.0 release. Would it be possible to get an update?

Thanks for all the hard work! Wallabag has been great for me for the past several years :)

**Reply by j0k3r** [reply]
Hello @dmhowcroft, 

At the moment, we are mainly focused on releasing [2.7.0](https://github.com/wallabag/wallabag/milestone/79) with some important technical update (maybe a 2.6 patch will also be released).
Then we'll have more technical stuff for the [2.8.0](https://github.com/wallabag/wallabag/milestone/86)

All these technical upgrade are required to have a more cleaned codebase before starting to properly work on the [3.0](https://github.com/wallabag/wallabag/milestone/44). So my guess is that it won't come until ends of 2025

**Reply by dmhowcroft** [reply]
Merci, @j0k3r, good to know!
